### PR TITLE
Added conversion for two-dimensional arrays

### PR DESCRIPTION
**Pull request by sebasguts**

### CODE REVIEW

File: pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl [hunk 1]
```diff
@@ -202,6 +202,56 @@ function gap_to_julia( ::Type{Array{Union{Nothing,T},1}}, obj :: MPtr, recursion
...
+    end
+    if haskey(recursion_dict, obj)
+        return recursion_dict[obj]
+    end
+    len_list_outer = length(obj)
+    len_list_inner = len_list_outer == 0 ? 0 : length(obj[1])
+    new_array = Array{T,2}( undef, len_list_outer, len_list_inner )
+    recursion_dict[obj] = new_array
+    for i in 1:len_list_outer
+        for j in 1:len_list_outer
```
Comment (ThomasBreuer) [inline pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl]: `1:len_list_inner`

File: pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl [hunk 2]
```diff
@@ -202,6 +202,56 @@ function gap_to_julia( ::Type{Array{Union{Nothing,T},1}}, obj :: MPtr, recursion
...
+    end
+    if haskey(recursion_dict, obj)
+        return recursion_dict[obj]
+    end
+    len_list_outer = length(obj)
+    len_list_inner = len_list_outer == 0 ? 0 : length(obj[1])
+    new_array = Array{Union{Nothing,T},2}( undef, len_list_outer, len_list_inner )
+    recursion_dict[obj] = new_array
+    for i in 1:len_list_outer
+        for j in 1:len_list_outer
```
Comment (ThomasBreuer) [inline pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl]: `1:len_list_inner`

**Comment by fingolfin** [comment]
> as far as I see, equal subobjects on the source side are turned into identical objects on the target side.

What makes you say that? This would be a bug in general. Do you have some specific examples for this?

File: pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl (hunk 1 above)
Comment (sebasguts) [reply pkg/GAPJulia/JuliaInterface/julia/gap_to_julia.jl]: thank you for spotting this, fixed now

**Comment by sebasguts** [comment]
I think this is a misunderstanding. Identical objects on the source side are turned into identical objects (on purpose). This is done by using `IdDict()`, which compares by `===`, not `==`, for caching already converted objects.

**Comment by ThomasBreuer** [comment]
Thanks for the explanation.

**Pull request merged**